### ISSUE TITLE
[Suggestion] Fireflies

### ISSUE THREAD

**Issue by GoobyMain** [RandomMcSomethin/fallingleaves#32]
I know the main idea of the mod is falling leaves, but I feel like firefly particles could be implemented in a very similar way and add really nice ambience to swamps at night.

It makes sense to me for fireflies to hover around directly under the canopy of trees, so I assume a lot of code for the falling leaves particles in this mod could be reused and little firefly particles could be integrated really well

I imagine them only spawning under oak or mangrove leaves in swamps during night, and only lasting a couple seconds, to make it seem as if the fireflies' lights are fading on and off. They could probably get away with being a single yellow pixel (no texture work), and maybe the particle could be emissive, so you don't even have to deal with any lighting but could still get a glowing effect

**Comment by Fourmisain** [comment]
Hm.. Not sure about this one, there's already [Illuminations](https://www.curseforge.com/minecraft/mc-mods/illuminations) which adds fireflies mainly in swamps at night - they're highly configurable too. The mod also adds a lot beyond that - which can be disabled if not wanted.

In real life, fireflies mostly tend to fly around the ground or sit on the grass. When they are around trees, they're usually all _over_ it, not under it.

So I'd say this is out of the scope of this mod, what do you think @BrekiTomasson, @RandomMcSomethin?

**Comment by RandomMcSomethin** [comment]
I think it's probably out of scope, especially since there are plenty of other firefly mods out there and I don't want to trample on anyone's feet.  Maybe as a separate mod if necessary, but I'd rather keep Falling Leaves about falling leaves.

**Comment by BrekiTomasson** [comment]
While this issue has been closed, I figured I should add my two cents anyway. While this is definitely a lovely idea, it does feel out of scope for this particular mod. However, I could well imagine us creating a mod that does *only* the fireflies in the way described above, as the suggested implementation is different enough from `illuminations` that it would fill a different use case... But again, not as part of this mod.
